### PR TITLE
Log HTTP error responses

### PR DESCRIPTION
**Pull request by MartinBasti**
Log response text from OCP
    
Added hook to requests to log response when a HTTP request failed.
    
However with retries it's tricky, if failure limit is reached, it will
just fail withou passing response to hooks. To be able to process
request while using retries, Retry(raise_on_status=False) has to be
used.
    
However this param is available from urllib3 > 1.15, in RHEL7 there is
urllib3 ~= 1.10, so response text will be logged only with newer urllib3
version.
    
<s>Also Retry() rasises exception by default when a HTTP request was not
successful. To keep this behavior and not break current code, another
hook has been added to raise exception when request was not successful.</s>
(no need for this, code checks responses and this change would even break unittests)
    
* CLOUDBLD-2800

Signed-off-by: Martin Bašti <mbasti@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

**Comment by MartinBasti** [comment]
PR 1000 :1st_place_medal:

**Review by mkosiarc — APPROVED** [review]

File: tests/test_http.py [hunk 1]
```diff
@@ -26,6 +28,12 @@ def s():
     return HttpSession(verbose=True)
 
 
+@pytest.fixture(scope='module')
+def error_response_logged():
+    """Error response loggin works only with urllib3>1.15"""
```
Comment (ben-alkov) [inline tests/test_http.py:33]: Typo: "loggin"

File: osbs/http.py [hunk 2]
```diff
@@ -75,6 +77,18 @@ def request(self, url, *args, **kwargs):
...
 
+def make_retry(**kwargs):
+    """Make initialized Retry object based on urllib3 version
+
+    :param kwargs: kwargs acceptable by urllib3.util.Retry class
+    :return: urllib3.util.Retry object
+    """
+    if LooseVersion(urllib3_version) < LooseVersion('1.15'):
+        # `raise_on_status` is not supported with older versions of urllib3 (RHEL7)
+        kwargs.pop('raise_on_status', None)
```
Comment (ben-alkov) [inline osbs/http.py:88]: Clever :wink:

**Review by lkolacek — APPROVED** [review]

File: tests/test_http.py (hunk 1 above)
Comment (MartinBasti) [reply tests/test_http.py]: fixed

**Review by ben-alkov — APPROVED** [review]

**Comment by MartinBasti** [comment]
Release notes:
- OCP error response content is logged

**Pull request merged**